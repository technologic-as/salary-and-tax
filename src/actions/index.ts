import Cookie from 'js-cookie';
import { updateIntl } from 'react-intl-redux';
import { change, submit } from 'redux-form';
import { form } from '../components/SalaryParameters';
import { getMessages } from '../translation';
import { Dispatch } from 'redux';
import { SupportedLocale } from '../constants';

export const FORM_CHANGE = 'FORM_CHANGE';

export const formChange = (values: any) => ({ type: FORM_CHANGE, ...values });
export type FormChange = { type?: string; [k: string]: unknown };

export const inputHasChanged =
  (values: { [k: string]: unknown } = {}) =>
  (dispatch: Dispatch) => {
    if (values.locale) {
      switchLanguage(values.locale as SupportedLocale)(dispatch);
    }
    return dispatch(formChange(values));
  };

export const switchLanguage = (locale: SupportedLocale) => (dispatch: Dispatch) => {
  Cookie.set('locale', locale);
  dispatch(updateIntl({ locale: locale, messages: getMessages(locale) }));
};

export const dividendsClick = (point: { x: number }) => (dispatch: Dispatch) => {
  dispatch(change(form, 'surplus.include', true));
  dispatch(change(form, 'surplus.amount', point.x, true));
  dispatch(submit(form));
};
