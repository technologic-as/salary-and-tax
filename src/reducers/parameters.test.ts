import { FORM_CHANGE } from '../actions';
import reducer from './parameters';

describe('parameters', () => {
  it('should have initial state', function () {
    expect(reducer()).toMatchSnapshot();
  });
  describe('FORM_CHANGE', () => {
    it('should update data', () => {
      expect(reducer({}, { type: FORM_CHANGE, changeMe: 'new' })).toMatchSnapshot();
    });
  });
});
