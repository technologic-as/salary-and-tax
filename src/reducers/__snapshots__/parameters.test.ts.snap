// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parameters FORM_CHANGE should update data 1`] = `
Object {
  "changeMe": "new",
  "type": "FORM_CHANGE",
}
`;

exports[`parameters should have initial state 1`] = `
Object {
  "cut": 10,
  "employerFee": Object {
    "include": true,
    "rate": 14.1,
  },
  "ensuringFee": 15000,
  "graph": Object {
    "increments": 10000,
  },
  "hourRate": 1100,
  "hoursPerYear": 1730,
  "locale": "nb",
  "pension": Object {
    "include": true,
    "step1": Object {
      "rate": 2,
      "threshold": Object {
        "high": 708991,
        "low": 99858,
      },
    },
    "step2": Object {
      "rate": 2,
      "threshold": Object {
        "high": 1198296,
        "low": 708991,
      },
    },
  },
  "surplus": Object {
    "amount": 0,
    "include": true,
    "sevenG": true,
  },
  "vacationSavings": Object {
    "include": true,
    "rate": 12,
  },
}
`;
